svm/tests/integration_test.rs

```diff
@@ -531,23 +625,206 @@ fn simple_transfer() -> Vec<SvmTestEntry> {
             system_instruction::transfer(&source, &Pubkey::new_unique(), transfer_amount);
         instruction.program_id = Pubkey::new_unique();
 
-        test_entry.transaction_batch.push(TransactionBatchItem {
-            transaction: Transaction::new_signed_with_payer(
+        test_entry.push_transaction_with_status(
+            Transaction::new_signed_with_payer(
                 &[instruction],
                 Some(&source),
                 &[&source_keypair],
                 Hash::default(),
             ),
-            asserts: TransactionBatchItemAsserts::not_executed(),
-            ..TransactionBatchItem::default()
-        });
+            ExecutionStatus::Discarded,
+        );
+    }
+
+    vec![test_entry]
+}
+
+fn simple_nonce_fee_only(enable_fee_only_transactions: bool) -> Vec<SvmTestEntry> {
```

Comment:
These tests are already fantastic but what do you think about covering the same-nonce-and-fee-payer case as well here?